src/core/lib/iomgr/ev_epollex_linux.cc

```diff
@@ -510,6 +529,18 @@ static grpc_error* pollable_add_fd(pollable* p, grpc_fd* fd) {
   static const char* err_desc = "pollable_add_fd";
   const int epfd = p->epfd;
 
+  for (int i = 0; i < MAX_PROBE_EPOLL_FDS; i++) {
+    int idx = (i + p->cache_cursor) % MAX_PROBE_EPOLL_FDS;
+    if (p->cache[idx].fd == fd->fd && p->cache[idx].salt == fd->salt) {
+      return GRPC_ERROR_NONE;
+    }
+  }
+  // Add to cache
+  int idx = (p->cache_cursor - 1) % MAX_PROBE_EPOLL_FDS;
```

Comment:
I think this is going to end up replacing the same element over and over in case of collision... it may be worth trying a randomized replacement algorithm

Reply:
I agree, this is kind of a wonky eviction algorithm.

I've switched to a LRU eviction algorithm, which should probably do good enough in most cases.